### DISCUSSION TITLE
Make a check whether an object should update or not depending on fields changing

### DISCUSSION THREAD

**Discussion by Voldemorten** [directus/directus#16202] [Feature Requests]
When you update an item through the ItemsService (https://github.com/directus/directus/blob/main/api/src/services/items.ts), you will trigger and update whether or not the actual object changes. 

For instance, I have an item like this `{id: 42, color: 'blue', last_updated: '2022-10-27 14:54:59'}` and I run `itemsService.updateOne(42, {color: 'blue')` it updates the item, and thus changes it to `{id: 42, color: 'blue', last_updated: '2022-10-27 14:56:39'}`. 

The reason why this is an issue, is that it creates a revision. If you have enough changes (that changes nothing) generated by a script for example and then you have a user changing something in the middle of these revisions, it's very hard to find the user generated revision. 

Also if you have the `date_updated` field enabled, you would always get a visual revision, such as this: 
![image](https://user-images.githubusercontent.com/8776117/198291024-ecf8b998-9404-456c-8efe-cb2c2dce49c4.png)

And with an example where you try to update an item, but all fields are the same, you would the revision like this:
![image](https://user-images.githubusercontent.com/8776117/198291157-7966d5e1-a11d-4c7b-b6f0-0d81e45cf67f.png)

Wouldn't it make sense to check that if nothing changes, then you shouldn't update the item?

**Comment by Voldemorten** [comment]
If people think that it's a good idea to do something about it, I might be able to work on it my self, if people can point me in the right direction. 

The code must be updated somewhere in the ItemsService.updateMany(...), but I can't quite figure out where? I'm guessing it's somewhere near line 567: `if (Object.keys(payloadWithTypeCasting).length > 0) {`.
And also this might be something to add to the mutationOpts as well. Something like `forceUpdateItems`. And maybe even default it to `true`. 
Anyone able to help me out?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
